src/test/java/org/aist/aide/generaltypeservice/unit/PatternCrudServiceTests.java

```diff
@@ -0,0 +1,199 @@
+package org.aist.aide.generaltypeservice.unit;
+
+import org.aist.aide.generaltypeservice.domain.exceptions.NotFoundException;
+import org.aist.aide.generaltypeservice.domain.exceptions.ValidationFailureException;
+import org.aist.aide.generaltypeservice.domain.models.Pattern;
+import org.aist.aide.generaltypeservice.domain.services.PatternCrudService;
+import org.aist.aide.generaltypeservice.service.repositories.PatternRepository;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class PatternCrudServiceTests {
+    @Mock
+    private PatternRepository patternRepository;
+
+    @InjectMocks
+    private PatternCrudService patternCrudService;
+
+    @Before
+    public void initMocks(){
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void givenAPatternExists_WhenRequestingThePattern_ThePatternIsReturned() throws NotFoundException {
+        // arrange
+        var type = "Phone";
```

Comment:
Maybe move these to a class wide final constant, or a class or constants to clean test a bit.(so many Phones)